hpat/compiler.py

```diff
@@ -150,145 +63,104 @@ def inline_calls(func_ir, _locals):
     # CFG simplification fixes this case
     func_ir.blocks = ir_utils.simplify_CFG(func_ir.blocks)
 
+#TODO: remove these helper functions when Numba provide appropriate way to manipulate passes
+def pass_position(pm, location):
+    assert pm.passes
+    pm._validate_pass(location)
+    for idx, (x, _) in enumerate(pm.passes):
+        if x == location:
+            return idx
+    else:
+        raise ValueError("Could not find pass %s" % location)
+
+
+def add_pass_before(pm, pass_cls, location):
+    assert pm.passes
+    pm._validate_pass(pass_cls)
+    position = pass_position(pm, location)
+    pm.passes.insert(position, (pass_cls, str(pass_cls)))
+
+    # if a pass has been added, it's not finalized
+    pm._finalized = False
+
+def replace_pass(pm, pass_cls, location):
+    assert pm.passes
+    pm._validate_pass(pass_cls)
+    position = pass_position(pm, location)
+    pm[position] = (pass_cls, str(pass_cls))
+
+    # if a pass has been added, it's not finalized
+    pm._finalized = False
+
+@register_pass(mutates_CFG=True, analysis_only=False)
+class InlinePass(FunctionPass):
+    _name = "hpat_inline_pass"
+
+    def __init__(self):
+        pass
```

Comment:
Looks like it should work the same way without the stub.

Reply:
```__init__``` is marked as an abstract method in the base class